assert/assertions.go

```diff
@@ -898,9 +898,7 @@ func ElementsMatch(t TestingT, listA, listB interface{}, msgAndArgs ...interface
 	aLen := aValue.Len()
 	bLen := bValue.Len()
 
-	if aLen != bLen {
-		return Fail(t, fmt.Sprintf("lengths don't match: %d != %d", aLen, bLen), msgAndArgs...)
-	}
+	var extraA, extraB []interface{}
```

Comment:
The function `ElementsMatch` looks a bit long and unwieldy. Would it make sense to refactor it into a few smaller, unit-tested functions?

Reply:
I have split `ElementsMatch` to smaller functions and added a test for the main one.